modules/stochastic_tools/src/surrogates/GaussianProcessTrainer.C

```diff
@@ -151,9 +204,198 @@ GaussianProcessTrainer::finalize()
   else
     _param_standardizer.set(0, 1, _n_params);
 
-  _covariance_function->buildHyperParamMap(_hyperparam_map, _hyperparam_vec_map);
   _K.resize(_training_params.rows(), _training_params.rows());
+
+#ifdef LIBMESH_HAVE_PETSC
+  if (_do_tuning)
+    hyperparamTuning();
+#endif
+
   _covariance_function->computeCovarianceMatrix(_K, _training_params, _training_params, true);
   _K_cho_decomp = _K.llt();
   _K_results_solve = _K_cho_decomp.solve(_training_data);
+
+  _covariance_function->buildHyperParamMap(_hyperparam_map, _hyperparam_vec_map);
+}
+
+#ifdef LIBMESH_HAVE_PETSC
+int
+GaussianProcessTrainer::hyperparamTuning()
```

Comment:
What is the purpose of the return value here?

Reply:
This was due to PETSc CHKERRQ behavior, I have switched to CHKERRV, which i think is appropriate in this situation. See here https://www.mcs.anl.gov/petsc/petsc-current/docs/manualpages/Sys/CHKERRQ.html#CHKERRQ

Reply:
I think we should self contain all the petsc stuff to this function. So keeping this function as void and having a `CHKERRQ` within in. What do you think?

Reply:
Update: Retained CHKERRQ and added an error check on hyperparamTuning() return